File: internal/domain/venue/venue.go
package venue

import "strings"

type Venue struct {
	ID          int64  `json:"id"`
	Name        string `json:"name"`
	CountryCode string `json:"countryCode"`
	City        string `json:"city"`
}

func NewVenue(name, countryCode, city string) (Venue, error) {
	return Venue{
		Name:        strings.TrimSpace(name),
		CountryCode: strings.TrimSpace(countryCode),
		City:        strings.TrimSpace(city),
	}, nil
}

File: frontend/src/lib/features/venues/venue.ts
import { baseFields } from "$lib/model";
import z from "zod";

export const venueSchema = z.object({
	...baseFields.shape,
	name: z.string().nonempty(),
	countryCode: z.string().nonempty(),
	city: z.string().nonempty()
});

export type Venue = z.infer<typeof venueSchema>;
